app/scripts/views/complete_sign_up.js

```diff
@@ -139,10 +141,21 @@ define(function (require, exports, module) {
             } else if (
                 AuthErrors.is(err, 'INVALID_VERIFICATION_CODE') ||
                 AuthErrors.is(err, 'INVALID_PARAMETER')) {
-              // These server says the verification code or any parameter is
-              // invalid. The entire link is damaged.
-              verificationInfo.markDamaged();
-              err = AuthErrors.toError('DAMAGED_VERIFICATION_LINK');
+
+              // When coming from sign-in confirmation verification, show a
+              // verification link expired error instead of damaged verification link.
+              // This error is generated because the link has already been used.
+              if (self.viewName === 'complete-signin') {
+                // Disable resending verification, can only be triggered from new sign-in
+                self._showResend = false;
```

Comment:
Instead of looking at this.viewName directly, I think you could also [check this.isSignIn](https://github.com/mozilla/fxa-content-server/blob/025ed9141a735a1395b9ae8a39f575b10a410171/app/scripts/views/mixins/verification-reason-mixin.js#L35).